src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -602,8 +639,57 @@ def post_process(self, r):
+        env_id = self.containerapp_def["properties"]['environmentId']
+        parsed_managed_env = parse_resource_id(env_id)
+        env_name = parsed_managed_env['name']
+        env_rg = parsed_managed_env['resource_group']
+
+        # Parse location
+        env_info = self.get_environment_client().show(cmd=self.cmd, resource_group_name=env_rg, name=env_name)
+        location = env_info['location']
+
+        # Set image to None if it was previously set to the default image (case where image was not provided by the user) else reformat it
+        image = None if self.get_argument_image().__eq__(HELLO_WORLD_IMAGE) else _reformat_image(self.get_argument_source(), self.get_argument_repo(), self.get_argument_image())
+
+        if not self.get_argument_source() or _has_dockerfile(self.get_argument_source(), dockerfile):
+            dockerfile_content = _get_dockerfile_content(self.get_argument_repo(), self.get_argument_branch(), token, self.get_argument_source(), self.get_argument_context_path(), dockerfile)
+            ingress, target_port = _get_ingress_and_target_port(self.get_argument_ingress(), self.get_argument_target_port(), dockerfile_content)
+        # Construct ContainerApp
+        resource_group = ResourceGroup(self.cmd, env_rg, location=location)
+        env = ContainerAppEnvironment(self.cmd, env_name, resource_group, location=location)
+        app = ContainerApp(self.cmd, self.get_argument_name(), resource_group, None, image, env, target_port, self.get_argument_registry_server(), self.get_argument_registry_user(), self.get_argument_registry_pass(), self.get_argument_env_vars(), self.get_argument_workload_profile_name(), ingress)
+
+        # Fetch registry credentials
+        _get_registry_details(self.cmd, app, self.get_argument_source())  # fetch ACR creds from arguments registry arguments
+
+        if self.get_argument_source():
+            # Uses buildpacks to generate image if Dockerfile was not provided by the user
+            app.run_acr_build(dockerfile, self.get_argument_source(), quiet=False, build_from_source=not _has_dockerfile(self.get_argument_source(), dockerfile))
+            # Update image
+            self.containerapp_def["properties"]["template"]["containers"][0]["image"] = HELLO_WORLD_IMAGE if app.image is None else app.image
+
+        if self.get_argument_repo():
+            # Get GitHub access token
+            token = get_token(self.cmd, self.get_argument_repo(), self.get_argument_token())
+            _create_github_action(app, env, self.get_argument_service_principal_client_id(), self.get_argument_service_principal_client_secret(),
+                              self.get_argument_service_principal_tenant_id(), self.get_argument_branch(), token, self.get_argument_repo(), self.get_argument_context_path())
+            cache_github_token(self.cmd, token, self.get_argument_repo())
+            r = self.client.show(cmd=self.cmd, resource_group_name=env_rg, name=app.name)
+            return r
```

Comment:
We return a Container App object in the case where the `--repo` argument was provided, but it doesn't appear we're returning thing in the case where the `--source` argument was provided -- is this expected? I haven't looked at where this function is being called and how the returned object is being used by the caller, so it may or may not matter, but I just wanted to iron out the differences between the two flows.

Reply:
`--source` is executed while creating the payload before creating the Container App [here](https://github.com/snehapar9/azure-cli-extensions/blob/snehapar/Add-source-to-containerapp-update/src/containerapp/azext_containerapp/containerapp_decorator.py#L573) and `--repo` is executed after creating the Container App [here](https://github.com/snehapar9/azure-cli-extensions/blob/snehapar/Add-source-to-containerapp-update/src/containerapp/azext_containerapp/containerapp_decorator.py#L1357).

We need to return a Container App object in case of `--repo` to make sure the image is updated. In case of `--source`,  we are updating the image [here](https://github.com/snehapar9/azure-cli-extensions/blob/snehapar/Add-source-to-containerapp-update/src/containerapp/azext_containerapp/containerapp_decorator.py#L668) and do not need to return any value because `self.containerapp_def` has a global scope.